controller/network/network.go

```diff
@@ -253,8 +253,9 @@ func (network *Network) ConnectRouter(r *Router) {
 	network.routerChanged <- r
 
 	for _, h := range network.routerPresenceHandlers {
-		h.RouterConnected(r)
+		go h.RouterConnected(r)
 	}
+	go network.ValidateTerminators(r)
```

Comment:
I looked at doing it, but decided it didn't make sense. Other places we use presence handlers either because the logic is optional or because it's non-fabric code. I figured making this into a presence handler would just obscure things more.